LangChain4j Agent MCP tests do not run properly on GitHub Windows runner

Follow up from https://github.com/apache/camel-quarkus/pull/8001.

There's something odd happening on the GitHub Windows runner with `Langchain4jAgentTest.agentWithMcpClient`. On a local Windows 11 machine, the test passes fine:

The `server-everything` process process is started and produces messages:

```
[dev.langchain4j.mcp.client.transport.stdio.StdioMcpTransport] (main) Starting process: [npx.cmd, -y, @modelcontextprotocol/server-everything]
[dev.langchain4j.mcp.client.transport.stdio.StdioMcpTransport] (main) PID of the started process: 1480
[MCP] (main) > {"jsonrpc":"2.0","id":0,"method":"initialize","params":{"protocolVersion":"2024-11-05","capabilities":{"roots":{"listChanged":true}},"clientInfo":{"name":"langchain4j","version":"1.0"}}}
[dev.langchain4j.mcp.client.transport.stdio.ProcessStderrHandler] (Thread-31) [ERROR] Starting default (STDIO) server...
[dev.langchain4j.mcp.client.transport.stdio.ProcessStderrHandler] (Thread-31) [ERROR] Starting logs update interval
```

On GitHub Actions, the process is apparently started, but it doesn't seem to do anything:

```
DEBUG [dev.langchain4j.mcp.client.transport.stdio.StdioMcpTransport] (main) Starting process: [npx.cmd, -y, @modelcontextprotocol/server-everything]
DEBUG [dev.langchain4j.mcp.client.transport.stdio.StdioMcpTransport] (main) PID of the started process: 1296
[MCP] (main) > {"jsonrpc":"2.0","id":0,"method":"initialize","params":{"protocolVersion":"2024-11-05","capabilities":{"roots":{"listChanged":true}},"clientInfo":{"name":"langchain4j","version":"1.0"}}}
```

--- a/integration-tests/langchain4j-agent/src/main/java/org/apache/camel/quarkus/component/langchain4j/agent/it/AgentProducers.java
+++ b/integration-tests/langchain4j-agent/src/main/java/org/apache/camel/quarkus/component/langchain4j/agent/it/AgentProducers.java
@@ -235,6 +235,8 @@ public class AgentProducers {
             return new AgentWithoutMemory(new AgentConfiguration()
                     .withChatModel(chatModel)
                     .withMcpClient(new DefaultMcpClient.Builder()
+                            // Startup on Windows is really slow (around 1min).
+                            .initializationTimeout(Duration.ofSeconds(120))
                             .transport(new StdioMcpTransport.Builder()
                                     .command(List.of(ProcessUtils.getNpxExecutable(), "-y",
                                             "@modelcontextprotocol/server-everything"))

--- a/integration-tests/langchain4j-agent/src/main/resources/application.properties
+++ b/integration-tests/langchain4j-agent/src/main/resources/application.properties
@@ -16,3 +16,6 @@
 ## ---------------------------------------------------------------------------
 quarkus.native.resources.includes=rag/*
 quarkus.http.test-timeout=120S
+
+# This can be handy for detecting output from DefaultMcpClient npx process in case of eg. slow Windows machine.
+# quarkus.log.category."dev.langchain4j.mcp.client.transport.stdio".level=DEBUG

--- a/integration-tests/langchain4j-agent/src/test/java/org/apache/camel/quarkus/component/langchain4j/agent/it/Langchain4jAgentTestResource.java
+++ b/integration-tests/langchain4j-agent/src/test/java/org/apache/camel/quarkus/component/langchain4j/agent/it/Langchain4jAgentTestResource.java
@@ -17,6 +17,8 @@
 package org.apache.camel.quarkus.component.langchain4j.agent.it;
 
 import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
@@ -27,7 +29,6 @@ import java.util.concurrent.TimeUnit;
 import com.github.tomakehurst.wiremock.stubbing.StubMapping;
 import org.apache.camel.quarkus.component.langchain4j.agent.it.util.ProcessUtils;
 import org.apache.camel.quarkus.test.wiremock.WireMockTestResourceLifecycleManager;
-import org.junit.jupiter.api.condition.OS;
 
 public class Langchain4jAgentTestResource extends WireMockTestResourceLifecycleManager {
     private static final String OLLAMA_ENV_URL = "LANGCHAIN4J_OLLAMA_BASE_URL";
@@ -93,20 +94,40 @@ public class Langchain4jAgentTestResource extends WireMockTestResourceLifecycleM
 
     private Boolean isNodeJSInstallationExists() {
         try {
-            // TODO: Suppress MCP tests in GitHub Actions for windows - https://github.com/apache/camel-quarkus/issues/8007
-            if (OS.current().equals(OS.WINDOWS) && System.getenv("CI") != null) {
+            ProcessBuilder pb = new ProcessBuilder()
+                    .command(ProcessUtils.getNpxExecutable(), "--version");
+            Process process = pb.start();
+
+            boolean finished = process.waitFor(20, TimeUnit.SECONDS);
+
+            if (!finished) {
+                process.destroyForcibly();
+                LOG.error("Command: %s took too long.".formatted(String.join(" ", pb.command())));
                 return false;
             }
 
-            Process process = new ProcessBuilder()
-                    .command(ProcessUtils.getNpxExecutable(), "--version")
-                    .start();
+            int exitCode = process.exitValue();
+            String output = readStream(process.getInputStream());
+            String errorOutput = readStream(process.getErrorStream());
 
-            process.waitFor(10, TimeUnit.SECONDS);
-            return process.exitValue() == 0;
+            if (exitCode == 0) {
+                LOG.info("Command: %s: %s".formatted(String.join(" ", pb.command()), output.trim()));
+            } else {
+                LOG.error("Command: %s failed with %s".formatted(String.join(" ", pb.command()), exitCode));
+                LOG.error("Process standard output %s".formatted(output.trim()));
+                LOG.error("Process error output %s".formatted(errorOutput.trim()));
+            }
+
+            return exitCode == 0;
         } catch (Exception e) {
             LOG.error("Failed detecting Node.js", e);
+            return false;
         }
-        return false;
+    }
+
+    private String readStream(InputStream inputStream) throws IOException {
+        try (inputStream) {
+            return new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+        }
     }
 }

